db/migrate/20240517145354_create_sign_in_certificates.rb

```diff
@@ -0,0 +1,15 @@
+class CreateSignInCertificates < ActiveRecord::Migration[7.1]
+  def change
+    create_table :sign_in_certificates do |t|
+      t.string :issuer, null: false
+      t.string :subject, null: false
+      t.string :serial, null: false
+      t.datetime :not_before, null: false
+      t.datetime :not_after, null: false
+      t.text :plaintext, null: false
+      t.references :configuration, polymorphic: true
```

Comment:
Should this have `null: false`?

Also, this is a nit but could we name this `config` since it's `ClientConfig` and `ServiceAccountConfig`

Reply:
TIL you can add `null: false` on a polymorphic reference column and it'll add the null constraint to both `*_type` and `*_id`.